python/vyos/ifconfig/interface.py

```diff
@@ -977,25 +977,76 @@ def del_tc_qdisc(self,dev,kind,handle):
             old_handle = rule['handle']
             old_kind = rule['kind']
             if old_dev == dev and old_handle == handle and old_kind == kind:
-                delete_tc_cmd = f'tc  qdisc del dev {dev} handle {handle} {kind}'
-                self._cmd(delete_tc_cmd)
+                if 'root' in rule and rule['root']:
+                    delete_tc_cmd = f'tc  qdisc del dev {dev} handle {handle} root {kind}'
+                    self._cmd(delete_tc_cmd)
+                else:
+                    delete_tc_cmd = f'tc  qdisc del dev {dev} handle {handle} {kind}'
+                    self._cmd(delete_tc_cmd)
 
 
 
     def apply_mirror(self,config):
+        # Please refer to the document for details
+        # https://man7.org/linux/man-pages/man8/tc.8.html
+        # https://man7.org/linux/man-pages/man8/tc-mirred.8.html
         ifname = config['ifname']
-
         # Remove existing mirroring rules
         self.del_tc_qdisc(ifname,'ingress','ffff:')
-
+        self.del_tc_qdisc(ifname,'prio','1:')
+        
         # Setting up packet mirroring
         mirror = dict_search('mirror', config)
         if mirror:
-            for interface in mirror:
+            two_way = False
+            direction = None
+            for interface, interface_conf in mirror.items():
+                if 'direction' not in interface_conf:
+                    two_way = True
+                    direction = None
+                    continue
+                else:
+                    if direction == None:
+                        direction = interface_conf['direction']
+                    else:
+                        if direction != interface_conf['direction']:
+                            two_way = True
+                            direction = None
+                            continue
+            
+            if two_way:
```

Comment:
Why having an extra "two way" control path? This is simply copied code from ingress and egress flows. Thus simplify the code path to:

```
if ingress:
  # code foo

if egress:
  # code bar
```

So you write the code only once.